resources/testharness.js

```diff
@@ -1820,21 +1832,31 @@
      * Assert that ``actual`` is a number greater than ``lower`` and less
      * than ``upper`` but not equal to either.
      *
-     * @param {number} actual - Test value.
-     * @param {number} lower - Number that ``actual`` must be greater than.
-     * @param {number} upper - Number that ``actual`` must be less than.
+     * @param {number|bigint} actual - Test value.
+     * @param {number|bigint} lower - Value that ``actual`` must be greater than.
+     * @param {number|bigint} upper - Value that ``actual`` must be less than.
      * @param {string} [description] - Description of the condition being tested.
      */
     function assert_between_exclusive(actual, lower, upper, description)
     {
         /*
-         * Test if a primitive number is between two others
+         * Test if a primitive number (or bigint) is between two others
          */
-        assert(typeof actual === "number",
+        assert(typeof actual === "number" || typeof actual === "bigint",
                "assert_between_exclusive", description,
                "expected a number but got a ${type_actual}",
                {type_actual:typeof actual});
 
+        assert(typeof lower === typeof upper,
+               "assert_between_exclusive", description,
+               "expected lower (${type_lower}) and upper (${type_upper}) types to match",
+               {type_lower:typeof lower, type_upper:typeof upper});
+
```

Comment:
Would you mind putting this before the `typeof actual` assert above, and add something like "test error"? It seems to me like this will never be the implementation's fault. (Also in `assert_between_inclusive`.)

Reply:
Done in c6b3b842fd